File: Algorithms/SelectionSort.py
# Driver Code
my_list = [923,6,234,56,3,0,123,5,7,3]

# length of the list
length = len(my_list)


for i in range(length):
    
    # index of the element we consider to have the least value
    min_index = i
    
    # i + 1 beacuse i th index is the one we are comparing
    for j in range(i + 1, length):
        
        # if the value at index(currently having the min value) is greater than the value..
        # ..present at another index in the same array/list 
        # then the min_index is changed to the index having smaller value
        if my_list[min_index] > my_list[j]:
            min_index = j
        
    # after one complete loop, the index having smallest value is identified (ignoring the smallest values which have been sorted before)
    # this smallest value is replaced by the index we considered to be smallest ( refer to line 11 : min_index = i)
    my_list[i], my_list[min_index] = my_list[min_index], my_list[i]
    
print("Sorted Array: ", my_list)

File: Algorithms/SelectionSort.vb
Module Module1

    Sub Main()
        Dim min_index, list_length, temp As Integer
        Dim my_list() As Integer

        'Driver Code
        my_list = {1, 0, 2, 7, 4, 3, 5, 6, 9, 8, 12, 54, 98, 909, 234, 87, 234, 89}

        'Length of the array
        list_length = my_list.Length


        For i = 0 To list_length - 1

            min_index = i

            For j = i + 1 To list_length - 1

                If my_list(min_index) > my_list(j) Then
                    min_index = j
                End If

            Next

            temp = my_list(i)
            my_list(i) = my_list(min_index)
            my_list(min_index) = temp

        Next

        For i = 0 To list_length - 1

            Console.WriteLine("my_list(" & Str(i) & ") = " & Str(my_list(i)))

        Next

        Console.ReadKey()

    End Sub

End Module
